httpx_ws/transport.py

```diff
@@ -108,8 +106,11 @@ async def write(
                 raise UnhandledWebSocketEvent(event)
 
     async def aclose(self) -> None:
-        await self.send({"type": "websocket.close"})
-        self.exit_stack.close()
+        with anyio.CancelScope(shield=True):
+            await self.send({"type": "websocket.close"})
+            await self._app_finished.wait()
+        # await self.exit_stack.aclose()  # This errors with:
+        # RuntimeError: Attempted to exit a cancel scope that isn't the current tasks's current cancel scope
```

Comment:
@agronholm Do I need to run this?

Reply:
Yes, closing the exit stack is absolutely crucial, and should be done in `__aexit__()`. If you're using task groups here, it doesn't make sense to have an `aclose()` method that can be called at point in time, because the cancel scope stack would possibly be corrupted then, as indicated by the error.